### ISSUE TITLE
NameError: global name 'prefix' is not defined

### ISSUE THREAD

**Issue by boutier** [fingon/pybabel#2]
```
$ sudo python3 babeld.py -d eth0
# ip -6 route flush proto 42
DEBUG:pybabel.babel:<pybabel.babel.BabelInterface object at 0xb6f4d4ac> queue_tlv Hello(interval=400, seqno=0) 2.0
DEBUG:__main__:<__main__.Timeout object at 0xb6ed9f4c> Timeout <bound method BabelInterface.tlv_send_timer of <pybabel.babel.BabelInterface object at 0xb6f4d4ac>>
DEBUG:__main__:<__main__.Timeout object at 0xb6ed9dcc> Timeout <bound method BabelInterface.hello_timer of <pybabel.babel.BabelInterface object at 0xb6f4d4ac>>
DEBUG:pybabel.babel:<pybabel.babel.BabelInterface object at 0xb6f4d4ac> queue_tlv RouteReq(plen=0, ae=0) 2.0
DEBUG:__main__:<__main__.Timeout object at 0xb6ed9f4c> Timeout.cancel
DEBUG:__main__:<__main__.Timeout object at 0xb6ed9f8c> Timeout <bound method BabelInterface.tlv_send_timer of <pybabel.babel.BabelInterface object at 0xb6f4d4ac>>
DEBUG:__main__:select dict_keys([<socket.socket object, fd=3, family=10, type=2, proto=0>]) 1.061603307723999
DEBUG:__main__:readable [<socket.socket object, fd=3, family=10, type=2, proto=0>]
DEBUG:pybabel.babel:process_tlvs fe80::21b:b1ff:fe83:8d35 - [Hello(interval=400, seqno=64106, l=6), IHU(interval=1200, body=b'\xe2\x91\xf5\xff\xfe\xccz\xbf', rxcost=292, ae=3, l=14), Update(metric=0, interval=1600, plen=128, omitted=0, ae=2, l=28, seqno=41267, body=b' \x01\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d2\x02\x00', flags=192), Update(metric=0, interval=1600, plen=128, omitted=16, ae=2, l=12, seqno=41267, body=b'\x02\x00', flags=128), Update(metric=0, interval=1600, plen=128, omitted=2, ae=2, l=26, seqno=41267, body=b'A\xd0\x00\x01\xf1\x9f\x02\x1b\xb1\xff\xfe\x83\x8d2\x02\x00', flags=128), Update(metric=0, interval=1600, plen=128, omitted=16, ae=2, l=12, seqno=41267, body=b'\x02\x00', flags=128), NH(body=b'\xc0\xa8\x04%', ae=1, l=6), Update(metric=0, interval=1600, plen=32, omitted=0, ae=1, l=16, seqno=41267, body=b'\xc0\xa8\x04%\x02\x00', flags=0), Update(metric=0, interval=1600, plen=32, omitted=0, ae=1, l=16, seqno=41267, body=b'\xc0\xa8\x04%\x02\x00', flags=0), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=13896, body=b'\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d;\x02\x01$', flags=192), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=13896, body=b'A\xd0\x00\x01\xf1\x9f\x02\x1b\xb1\xff\xfe\x83\x8d;\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=13896, body=b'\xc0\xa8\x04#\x02\x01$', flags=0), RID(rid=b'\x02\x1b\xb1\xff\xfe\x83\x8d>', l=10), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=10472, body=b'\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d\xb3\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=10472, body=b'A\xd0\x00\x01\xf1\x9f\x02\x1b\xb1\xff\xfe\x83\x8d\xb3\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=10472, body=b'\xc0\xa8\x04$\x02\x01$', flags=0), Update(metric=264, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=2258, body=b'\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d?\x02\x01$', flags=192), Update(metric=264, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=2258, body=b'A\xd0\x00\x01\xf1\x9f\x02\x1b\xb1\xff\xfe\x83\x8d?\x02\x01$', flags=128), Update(metric=264, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=2258, body=b'\xc0\xa8\x04&\x02\x01$', flags=0), RID(rid=b'\x1c\xfc,\r#\x7f e', l=10), Update(metric=556, interval=1600, plen=128, omitted=2, ae=2, l=28, seqno=57171, body=b'\x06`3\x01\x92\x08\xe0\x91\xf5\xff\xfe\xccz\x93\x02\x02$\x01', flags=128), Update(metric=556, interval=1600, plen=128, omitted=2, ae=2, l=28, seqno=57171, body=b'A\xd0\x00\x01\xf1\x9f\xe0\x91\xf5\xff\xfe\xccz\x93\x02\x02$\x01', flags=128), Update(metric=352, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=57171, body=b'\xc0\xa8\x04)\x02\x01$', flags=0), RID(rid=b'\xa0\x87\xd6\xd5dy\xd9\xe8', l=10), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=36080, body=b'\x06`3\x01\x92\x02\x00\x00\x00\x00\xac\x17$\x8a\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=128, omitted=7, ae=2, l=22, seqno=36080, body=b'\x08\xe2F\x9a\xff\xfeN\x91.\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=36080, body=b"\xc0\xa8\x04'\x02\x01$", flags=0), RID(rid=b'\xa2!\xb7\xff\xfe\xba\xdfW', l=10), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=19572, body=b'\x05\xc0\x15\x05k\x00\xa0!\xb7\xff\xfe\xba\xdfW\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=19572, body=b'A\xd0\x00\x01\xf1\x9f\xa0!\xb7\xff\xfe\xba\xdfW\x02\x01$', flags=128), Update(metric=256, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=19572, body=b'\xc0\xa8\x04!\x02\x01$', flags=0), Update(metric=256, interval=1600, plen=24, omitted=0, ae=1, l=16, seqno=19572, body=b'\xc0\xa8\x04\x02\x01$', flags=0), Update(metric=352, interval=1600, plen=128, omitted=2, ae=2, l=27, seqno=40840, body=b'\x06`3\x01\x92\x08\xb6\xb5/\xff\xfe\xb85\xc3\x02\x01$', flags=192), Update(metric=352, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=40840, body=b'\xc0\xa8\x04\x03\x02\x01$', flags=0), Update(metric=605, interval=1600, plen=128, omitted=2, ae=2, l=28, seqno=31823, body=b'A\xd0\x00\x01\xf1\x9f\xe2\x91\xf5\xff\xfe\xccz\x00\x02\x02$$', flags=192), Update(metric=605, interval=1600, plen=32, omitted=0, ae=1, l=18, seqno=31823, body=b'\xc0\xa8\x04\x1f\x02\x02$$', flags=0), RID(rid=b'\xe2\x91\xf5\xff\xfe\xccz\xbf', l=10), Update(metric=273, interval=1600, plen=128, omitted=15, ae=2, l=14, seqno=65192, body=b'\xbd\x02\x01$', flags=128), Update(metric=273, interval=1600, plen=32, omitted=0, ae=1, l=17, seqno=65192, body=b'\xc0\xa8\x04\x1e\x02\x01$', flags=0)]
DEBUG:pybabel.babel:<pybabel.babel.BabelNeighbor object at 0xb6ee5dec> process_hello
DEBUG:pybabel.babel:<pybabel.babel.Babel object at 0xb6f4698c> Babel.route_selection
DEBUG:pybabel.babel: remote routes: {}
DEBUG:pybabel.babel:<pybabel.babel.BabelNeighbor object at 0xb6ee5dec> process_ihu IHU(interval=1200, body=b'\xe2\x91\xf5\xff\xfe\xccz\xbf', rxcost=292, ae=3, l=14)
DEBUG:pybabel.babel: wrong target fe80::e291:f5ff:fecc:7abf != fe80::a00:27ff:fe8a:fbe8
Traceback (most recent call last):
  File "babeld.py", line 191, in <module>
    sys.loop()
  File "babeld.py", line 97, in loop
    self.readers[fd]()
  File "babeld.py", line 142, in _f
    babel.process_inbound(ifname, a, data)
  File "/home/boutier/programmation/networking/pybabel/pybabel/babel.py", line 423, in process_inbound
    self.interface(ifname).process_tlvs(address, p.tlvs)
  File "/home/boutier/programmation/networking/pybabel/pybabel/babel.py", line 320, in process_tlvs
    rid = prefix.address.packed[-RID_LEN:]
NameError: global name 'prefix' is not defined
```

If you need the tcpdump's dump, just ask for.

**Comment by fingon** [comment]
Fixed in 8766c57e2624108c0cbd12e7f2b1643f05690ff4 - was not covered by unit tests as they talk mostly to itself (and it does not use 0x40 update flag), and I broke it when doing the on-plane-fixes ;)

**Comment by boutier** [comment]
:-)
